esphome/components/hc8/hc8.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/core/automation.h"
+#include "esphome/components/sensor/sensor.h"
+#include "esphome/components/uart/uart.h"
+
+#include <cinttypes>
+#include <span>
+
+namespace esphome {
+namespace hc8 {
```

Comment:
```suggestion
namespace esphome::hc8 {
```
We are moving all new components to this style, can update both cpp and h files